src/components/connector.jl

```diff
@@ -6,12 +6,14 @@ using Mimi
     output =  Variable(index = [time])
 
     function run_timestep(p, v, d, ts)
-        if hasvalue(p.input1, ts)
+        try 
             v.output[ts] = p.input1[ts]
-        elseif hasvalue(p.input2, ts)
-            v.output[ts] = p.input2[ts]
-        else
-            error("Neither of the inputs to ConnectorCompVector have data for the current timestep: $(gettime(ts)).")
+        catch 
```

Comment:
@rjplevin @davidanthoff Now that I've added the new error to `getindex(timesteparray, timestep)` in the case of missing values, ConnectorComp relies on try/catch statements to determine if the shorter data is missing and if it should instead use the backup data. Is there a better way to make sure it's only catching that specific error, besides comparing the string of the error message? 

Alternatively, instead of relying on try/catch statements, I could add new parameters to the @defcomp that indicate the first and/or last years to use the shorter data for. 

```
@defcomp ConnectorCompVector begin
    input1 = Parameter(index = [time])
    input2 = Parameter(index = [time])
    output =  Variable(index = [time])
    short_first = Parameter()
    short_last = Parameter()

    function run_timestep(p, v, d, ts)
        if gettime(ts) >= short_first && gettime(ts) <= short_last
            v.output[ts] = p.input1[ts]
        else
            v.output[ts] = p.input2[ts]
        end
    end
end
```
These parameter values would be set when the connectorcomps are added during build, based on the `first` and `last` values of the source component. 

Which implementation would you prefer?